### PR TITLE
Backport PHP Compatibility fixes

### PR DESCRIPTION
**Pull request by mattyrob**
## Description
This PR attempts to fix PHPCompatability tests using backports instead of:
https://github.com/ClassicPress/ClassicPress/pull/766

## Motivation and context
The aim is to enable a test rather than allow failure on running.

## How has this been tested?
Tests will run as part of PR for assurance

## Screenshots
N/A

## Types of changes
- Bug fix / enhancement

### CODE REVIEW

File: src/wp-includes/functions.php [hunk 1]
```diff
@@ -175,10 +175,12 @@ function wp_maybe_decline_date( $date ) {
 	 * translate this to 'on'. Do not translate into your own language.
 	 */
 	if ( 'on' === _x( 'off', 'decline months names: on or off' ) ) {
+
+		$months          = $wp_locale->month;
+		$months_genitive = $wp_locale->month_genitive;
+
```
Comment (joyously) [inline src/wp-includes/functions.php:181]: This moving of two lines is not part of the 45611 changeset. I think it's from the big date changes done in WP 5.3.

File: src/wp-includes/functions.php [hunk 2]
```diff
@@ -190,6 +192,19 @@ function wp_maybe_decline_date( $date ) {
...
+			foreach ( $months as $key => $month ) {
+				$months[ $key ] = '#' . $month . ' (\d{1,2})(st|nd|rd|th)?#u';
+			}
+
+			foreach ( $months_genitive as $key => $month ) {
+				$months_genitive[ $key ] = '$1 ' . $month;
+			}
+
+			$date = preg_replace( $months, $months_genitive, $date );
+		}
```
Comment (joyously) [inline src/wp-includes/functions.php:207]: This whole section is not part of 45611.
See https://core.trac.wordpress.org/changeset/45611/trunk/src/wp-includes/functions.php

File: src/wp-includes/functions.php [hunk 3]
```diff
@@ -3322,10 +3334,13 @@ function _wp_json_prepare_data( $data ) {
  * @param int   $status_code The HTTP status code to output.
  */
 function wp_send_json( $response, $status_code = null ) {
-	@header( 'Content-Type: application/json; charset=' . get_option( 'blog_charset' ) );
+	if ( ! headers_sent() ) {
+		header( 'Content-Type: application/json; charset=' . get_option( 'blog_charset' ) );
 	if ( null !== $status_code ) {
```
Comment (joyously) [inline src/wp-includes/functions.php:3339]: This if statement should be indented under the new if added by the PR.

Comment (mattyrob) [reply src/wp-includes/functions.php:3339]: Indentation is incorrect, will update the PR.

File: src/wp-includes/functions.php (hunk 1 above)
Comment (mattyrob) [reply src/wp-includes/functions.php:181]: This is part of 45555 also.

File: src/wp-includes/functions.php [hunk 4]
```diff
@@ -134,7 +134,7 @@ function date_i18n( $dateformatstring, $unixtimestamp = false, $gmt = false ) {
 			}
 		}
 	}
-	$j = @date( $dateformatstring, $i );
+	$j = gmdate( $dateformatstring, $i );
```
[the remark this answers is not in the document]
Comment (joyously) [reply src/wp-includes/functions.php:137]: I think so. Apparently, we are doing the two changes in a different order.

**Pull request merged**